autogen/oai/gemini.py

```diff
@@ -0,0 +1,316 @@
+"""Create a OpenAI-compatible client for Gemini features.
+
+Resources:
+- https://ai.google.dev/docs
+- https://cloud.google.com/vertex-ai/docs/generative-ai/migrate-from-azure
+- https://blog.google/technology/ai/google-gemini-pro-imagen-duet-ai-update/
+- https://ai.google.dev/api/python/google/generativeai/ChatSession
+"""
+
+from __future__ import annotations
+
+import base64
+import os
+import random
+import re
+import time
+import warnings
+from io import BytesIO
+from typing import Any, Dict, List, Mapping, Union
+
+import google.generativeai as genai
+import requests
+from google.ai.generativelanguage import Content, Part
+from google.api_core.exceptions import InternalServerError
+from openai.types.chat import ChatCompletion
+from openai.types.chat.chat_completion import ChatCompletionMessage, Choice
+from openai.types.completion_usage import CompletionUsage
+from PIL import Image
+
+# from autogen.agentchat.contrib.img_utils import _to_pil, get_image_data
+from autogen.token_count_utils import count_token
+
+
+class GeminiClient:
+    """Client for Google's Gemini API.
+
+    TODO: this Gemini implementation does not support the following features yet:
```

Comment:
I wonder if it is worth tabulating and expanding on each limitation against each API outside of code and in `docs`.